src/route-data.ts

```diff
@@ -1,13 +1,18 @@
-export const RouteData = function (annotation?: string): any {
+export const RouteData = function (annotation?: string, useSnapshot?: boolean): any {
```

Comment:
Better to use an object as the second argument for backwards compatibility, it makes adding properties easier in the future 

    export const RouteData = function (annotation?: string, config?: RouteXxlConfig) { 
        ... 
    }

with RouteXxlConfig being an interface. Furthermore, thinking about it, this decorator can now assign an `Observable` or a `string` (if `useSnapshot === true`) to its property. But I think it would be better that when the URL changes the property is updated to, something like this

    while (parent && !data) {
        const targetKeyName = annotation || key.replace(/\$$/, '');

        data = parent.data.map(d => d[targetKeyName]);

        if (!config.observable) { 
            target[key] = parent.data[targetKeyName];      // initial value
            data.subscribe(val => target[key] = val);          // new values over time
        }
    ... 

  And use it like

    @RouteData('item', {observable: false}) item: string;

Hopefully this makes any sense :) and is still in line with what you want/need! What do you think?

Reply:
It makes sense, but making the data automatically update when the URL changes could be confusing to users, since it's different from how the Angular router does it. It shouldn't be a problem if we make it clear what's going on in the docs, though. I'll work on it tomorrow.

Reply:
I've hacked it into a [stackblitz](https://stackblitz.com/edit/angular-1znv1y) to play a bit with the idea. I still think its a nice feature. If for some reason someone can make a good point why this is a problem, we can always add an other property to prevent this behaviour, right?